lib/faraday/options/env.rb

```diff
@@ -52,12 +53,13 @@ module Faraday
   #
   # @!attribute reason_phrase
   #   @return [String]
-  class Env < Options.new(:method, :request_body, :url, :request,
-                          :request_headers, :ssl, :parallel_manager, :params,
-                          :response, :response_headers, :status,
-                          :reason_phrase, :response_body)
-
+  # rubocop:disable Metrics/BlockLength
+  Env = Options.new(:method, :request_body, :url, :request,
+                    :request_headers, :ssl, :parallel_manager, :params,
+                    :response, :response_headers, :status,
+                    :reason_phrase, :response_body) do
     # rubocop:disable Naming/ConstantName
+    # rubocop:disable Lint/ConstantDefinitionInBlock
```

Comment:
This rubocop lint is actually correct, switching from class definition to constant assignment is not a compatible change because constants gets defined in the global scope as opposed to the local one.

See this example:
```ruby
Test = Struct.new(:name) do
  CONSTANT = 1
end

class AnotherTest < Struct.new(:name)
  CONSTANT = '1'
end

CONSTANT #=> 1
Test::CONSTANT #=> uninitialized constant Test::CONSTANT (NameError)
AnotherTest::CONSTANT #=> "1"
```

As suggested by the cop, to maintain the same behaviour we need to use `const_set`:

```ruby
FixedTest = Struct.new(:name) do
  const_set(:CONSTANT, :one)
end

FixedTest::CONSTANT #=> :one
```